crates/bitwarden-exporters/src/encrypted_json.rs

```diff
@@ -0,0 +1,246 @@
+use base64::{engine::general_purpose::STANDARD, Engine};
+use bitwarden_crypto::{generate_random_bytes, Kdf, MasterKey};
+use serde::Serialize;
+use thiserror::Error;
+use uuid::Uuid;
+
+use crate::{
+    json::{self, export_json},
+    Cipher, Folder,
+};
+
+#[derive(Error, Debug)]
+pub enum EncryptedJsonError {
+    #[error(transparent)]
+    JsonExport(#[from] json::JsonError),
+
+    #[error("JSON error: {0}")]
+    Serde(#[from] serde_json::Error),
+
+    #[error("Cryptography error, {0}")]
+    Crypto(#[from] bitwarden_crypto::CryptoError),
+}
+
+pub(crate) fn export_encrypted_json(
+    folders: Vec<Folder>,
+    ciphers: Vec<Cipher>,
+    password: String,
+    kdf: Kdf,
+) -> Result<String, EncryptedJsonError> {
+    let decrypted_export = export_json(folders, ciphers)?;
+
+    let (kdf_type, kdf_iterations, kdf_memory, kdf_parallelism) = match kdf {
+        Kdf::PBKDF2 { iterations } => (0, iterations.get(), None, None),
+        Kdf::Argon2id {
+            iterations,
+            memory,
+            parallelism,
+        } => (
+            1,
+            iterations.get(),
+            Some(memory.get()),
+            Some(parallelism.get()),
+        ),
+    };
+
+    let salt: [u8; 16] = generate_random_bytes();
+    let salt = STANDARD.encode(salt);
+    let key = MasterKey::derive(password.as_bytes(), salt.as_bytes(), &kdf)?;
```

Comment:
We used to call this PIN, so open ended question is if we should use master key here, or define a new primitive.